src/features/events_list/components/FullState/FullState.tsx

```diff
@@ -0,0 +1,91 @@
+import { Virtuoso } from 'react-virtuoso';
+import { useAtom } from '@reatom/react';
+import { useEffect, useRef } from 'react';
+import { LoadingSpinner } from '~components/LoadingSpinner/LoadingSpinner';
+import { ErrorMessage } from '~components/ErrorMessage/ErrorMessage';
+import { i18n } from '~core/localization';
+import { AppFeature } from '~core/auth/types';
+import { createStateMap } from '~utils/atoms';
+import { eventListResourceAtom } from '~features/events_list/atoms/eventListResource';
+import { userResourceAtom } from '~core/auth';
+import { FeedSelector } from '../FeedSelector/FeedSelector';
+import { EpisodeTimelineToggle } from '../EpisodeTimelineToggle/EpisodeTimelineToggle';
+import { BBoxFilterToggle } from '../BBoxFilterToggle/BBoxFilterToggle';
+import { EventListSettingsRow } from '../EventListSettingsRow/EventListSettingsRow';
+import { EventCard } from '../EventCard/EventCard';
+import s from './FullState.module.css';
+
+export function FullState({
+  currentEventId,
+  onCurrentChange,
+}: {
+  currentEventId?: string | null;
+  onCurrentChange: (id: string) => void;
+}) {
+  const [{ data: eventsList, error, loading }] = useAtom(eventListResourceAtom);
+  const [{ data: userModel }] = useAtom(userResourceAtom);
+  const virtuoso = useRef(null);
+
+  const statesToComponents = createStateMap({
+    error,
+    loading,
+    data: eventsList,
+  });
+
+  // Virtual event list rendering effect
+  useEffect(() => {
+    // type any is used because virtuoso types doesn't have scrollToIndex method, but it's described in docs https://virtuoso.dev/scroll-to-index
+    const ref: any = virtuoso.current;
```

Comment:
suggest to remove any and comment and use 
```
const virtuoso = useRef<VirtuosoHandle>(null);
```
like here https://virtuoso.dev/virtuoso-api-reference/

Reply:
wow, they've updated types, cool!